internal/cache/memcache/mem_cache.go

```diff
@@ -0,0 +1,71 @@
+package memcache
+
+import (
+	"context"
+	"sync"
+
+	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/RHEnVision/provisioning-backend/internal/version"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type cache struct {
+	appTypeId      *string
+	appTypeIdMutex sync.Mutex
+	accountId      *Cache[accountKey, *models.Account]
+}
+
+type accountKey struct {
+	OrgID         string
+	AccountNumber string
+}
+
+func New() *cache {
+	c := cache{
+		accountId: newMemoryCache[accountKey, *models.Account](config.Application.Cache.Memory.CleanupInterval),
+	}
+
+	_ = promauto.NewGaugeFunc(prometheus.GaugeOpts{
+		Name:        "app_cache_account_items",
+		Help:        "The total number of cache items for account ID",
+		ConstLabels: prometheus.Labels{"service": version.PrometheusLabelName},
+	}, func() float64 {
+		return float64(c.accountId.Count())
+	})
```

Comment:
Just to make sure after reading the documentation
the purpose here is to create an “info” style metric with a constant value of 1 for The total number of cache items for account ID?

Reply:
Constant value? No, this gauge is supposed to contain total amount of accounts currently held in memory. So that could be 0, 1, thousands.

Reply:
yes, paste it from documentation but it means the a number not just 0

Reply:
Oh I see, no, this is a normal "gauge" measurement that counts all items in the account cache. It has nothing to do with that constant metric, the documentation just mention that this is also one of other uses of a gauge metric (to return constant only to be able to use labels to return some informative values like application version etc). We don't do that, the only label we return here is the application service name (which is "provisioning").